src/uct/api/uct.h

```diff
@@ -1310,15 +1316,25 @@ typedef struct uct_md_mem_attr {
      * Index of the system device on which the buffer resides. eg: NUMA/GPU
      */
     ucs_sys_device_t  sys_dev;
+
+    /**
+     * Base address of the allocation to which the provided buffer belongs to.
+     */
+    void              *base_address;
```

Comment:
1. pls add UCT level gtest for the new API. it has to pass on all transports with some basic implementation.
2. let's split UCP part to another PR.

Reply:
@yosefe removed UCP bits and extended existing gtest. Can you have a look when possible?